components/capi/include/locale.h

```diff
@@ -0,0 +1,14 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#pragma once
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+// opaque
+typedef struct Locale Locale;
```

Comment:
Since C does not have namespaces for imported headers, I would give all of these a manual namespace.  ICU4C prefixes all of its types with "U", like "ULocale".  Some options for us:

- XLocale (but "X" seems awfully generic)
- UXLocale (but "UX" means "user experience")
- ICU4X_Locale (I probably favor something more like this)
- icu4x_Locale (use lowercase even for type names?)

Below, for the function, I would probably do `icu4x_locale_create()`.

Reply:
Maybe Icu4x_ for types and icu4x_ for functions?

Reply:
Good call! This also means that internally we can stop doing `use ErasedDataProvider as ICUErasedDataProvider` only _some_ of the times